### ISSUE TITLE
Question: Running S3FS in a container without privilege mode.

### ISSUE THREAD

**Issue by downright-ux** [s3fs-fuse/s3fs-fuse#1246]
Hey I was wondering if there is an easy work around to get this working in none privilege mode.

**Comment by rhysmeister** [comment]
There's an implementation here using --cap-add SYS_ADMIN

https://hub.docker.com/r/efrecon/s3fs

That's still pretty open though.

**Comment by gaul** [comment]
Can someone explain this issue more fully?  Does s3fs require more permissions because of FUSE or some other issue that we can address?

**Comment by syalanurag1991** [comment]
@gaul Yes. In order to to mount an s3 bucket inside a docker container, one has to run docker run in privileged mode,
something like this `docker run -it --privileged my_docker_img /bin/bash`

else it will throw 'fuse: device not found, try 'modprobe fuse' first' error and won't mount the bucket.

**Comment by gaul** [comment]
I'm not sure if there is anything we can do here.  Do other FUSE filesystems like sshfs require the same permission?

**Comment by fly3366** [comment]
If use k8s, try device plugin for fuse

**Comment by EgorSemenov** [comment]
@fly3366  @syalanurag1991 @rhysmeister guys any updates? I am trying to mount bucket inside container inside openshift

**Comment by syalanurag1991** [comment]
Sorry, nope. I was not able to mount s3 without docker being previleged. And Sagemaker does not allow privileged dockers to be deployed. So had to find a workaround - basically download data to memory. Not an ideal system TBH.

**Comment by EgorSemenov** [comment]
@syalanurag1991 thanks for your answer

**Comment by gaul** [comment]
Sorry I lack background in this but this may be a Docker/container limitation.  You might want to look at:

https://github.com/docker/for-linux/issues/321#issuecomment-677744121

**Comment by acidjazz** [comment]
This would be huge since AWS Fargate does not allow priviledged mode - this library is not useable in fargate unless we can figure this out! 

https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/fargate-security-considerations.html

**Comment by leryn1122** [comment]
What's the minimum linux capability set when using s3fs as k8s sidecar or csi driver developping? SYS_ADMIN seems to be dangerous and overloaded.
  - https://man7.org/linux/man-pages/man7/capabilities.7.html

**Comment by ggtakec** [comment]
@leryn1122 
Even if you write a CSI driver, I think you need privileges in sidecar.
I plan to work on this issue in the future (hopefully FUSE3 support will fix everything), but I haven't started to it yet.

**Comment by ianrtracey** [comment]
Also running into this problem

**Comment by MMahendravarman** [comment]
Hi,

was anyone able to find solution for this?

Thanks
